### PR TITLE
Enable MKL-DNN in pip packages

### PR DESCRIPTION
**Pull request by TaoLv**
## Description ##
Discussion: https://lists.apache.org/thread.html/1a22dbd79098adab6d02d16e8d607bae2acc908c0bb1b085d28a51ba@%3Cdev.mxnet.apache.org%3E

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by TaoLv** [comment]
@lanking520 @perdasilva

**Comment by TaoLv** [comment]
I'm not sure whether the backward compatibility matters for convenient binary releases or not. What about if downstream projects explicitly call `pip install mxnet-cu90mkl` in their scripts [1]? 
@pengzhao-intel suggested to keep these *mkl variants till 2.0 major release [2].

[1]https://github.com/dmlc/gluon-cv/blob/master/tests/py3.yml#L19
[2]https://lists.apache.org/thread.html/87282a60cd73257d3dd4b57401ffe2eabd293d77456d9966b1f91336@%3Cdev.mxnet.apache.org%3E

**Comment by pengzhao-intel** [comment]
@szha does this make sense to keep *mkl variants for a short time? Or we need to remove from now.
@marcoabreu @samskalicky @larroy  any comments?

I am going to merging the PR in the early of next week in case there is no objection.

**Comment by szha** [comment]
Not really. Given that the content would be the same, having both would just waste space.

**Comment by wkcn** [comment]
I think keeping *mkl is better, since users may use *mkl to install MXNet.
To save space, we can add a dummy package, whose dependency is MXNet package, and raise the deprecated warning.

**Comment by pengzhao-intel** [comment]
> Not really. Given that the content would be the same, having both would just waste space.

@szha  @wkcn Agree. We will try if dummy package works now.

**Comment by TaoLv** [comment]
For these `.mk` files, I `git rm` those non-mkl variants and then `git mv` those mkl variants to non-mkl.

**Comment by TaoLv** [comment]
@szha @perdasilva Could you please review? Is it possible to verify these changes in dev CD? Thanks.

File: R-package/src/im2rec.h [hunk 1]
```diff
@@ -29,6 +29,12 @@
 
 #include <Rcpp.h>
 #include <string>
+#include <opencv2/core/version.hpp>
```
Comment (larroy) [inline R-package/src/im2rec.h]: Is this change related or is your branch out of date?

**Comment by larroy** [comment]
Can you rebase?

**Comment by TaoLv** [comment]
> Can you rebase?

Done.

**Comment by TaoLv** [comment]
@szha @perdasilva @leezu @larroy @pengzhao-intel Could you please review? How can we proceed this?

**Comment by samskalicky** [comment]
@TaoLv can you also update the cu102.mk files:
https://github.com/apache/incubator-mxnet/blob/master/make/staticbuild/linux_cu102.mk
https://github.com/apache/incubator-mxnet/blob/master/make/staticbuild/linux_cu102mkl.mk

**Comment by TaoLv** [comment]
Thank you @samskalicky. I over looked these files when merging the master branch. Will update them.

File: tools/staticbuild/README.md [hunk 2]
```diff
@@ -45,6 +44,7 @@ CMAKE_STATICBUILD=1 tools/staticbuild/build.sh cpu
 
 For the CMake build, you need to install `patchelf` first, for example via `apt
 install patchelf` on Ubuntu systems.
+>>>>>>> ae145cdb1de95e8275dd087585f4aab69bc94005
```
Comment (szha) [inline tools/staticbuild/README.md]: seems to be a leftover from resolving merge conflict

Comment (TaoLv) [reply tools/staticbuild/README.md]: Thanks. It's fixed.

**Comment by TaoLv** [comment]
Thank you @samskalicky . See https://github.com/apache/incubator-mxnet/issues/17560 for the website faliure.

**Comment by TaoLv** [comment]
@szha @leezu @samskalicky Is it good to go? Let me know if anything need be adjusted. Thanks.

**Comment by leezu** [comment]
@TaoLv Sheng just merged fixes for the CD pipeline. Let's wait 24 hours to verify CD works on master branch. Then we can merge this PR.

**Comment by szha** [comment]
Looks like the update has taken effect, though the jenkins jobs are still running for *mkl variants. http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/restricted-mxnet-cd%2Fmxnet-cd-release-job/detail/mxnet-cd-release-job/724/pipeline

**Comment by TaoLv** [comment]
@szha What can I for this situation? It seems we need change the value of `MXNET_VARIANTS` in the build environment.
